thetis/solver2d.py

```diff
@@ -331,6 +331,15 @@ def add_new_field(self, function, label, name, filename, shortname=None, unit='-
         :kwarg unit: units for field, e.g. '-'
         :kwarg preproc_func: optional pre-processor function which will be called before exporting
         """
+        assert isinstance(function, Function)
+        assert isinstance(label, str)
+        assert isinstance(name, str)
+        assert isinstance(filename, str)
+        assert shortname is None or isinstance(shortname, str)
+        assert isinstance(unit, str)
+        assert preproc_func is None or callable(preproc_func)
+        if label in self.fields:
+            print_output(f"Field '{label}' already exists. It will be overwritten.")
```

Comment:
While at it could you please add the equivalent to solver 3d as well?

Reply:
I did initially put it in as an assertion, but some of the tests crashed because they initialise twice (I think without meaning to). Happy to switch it and then modify those tests. Can do for 3D, too.

Reply:
Got it, the reason it was already in there is because the default 'tracer_2d' is hard-coded into `field_metadata`. Have removed it and made sure that it isn't in there when a solver object is created.